crates/ra_mbe/src/subtree_source.rs

```diff
@@ -447,39 +314,31 @@ fn convert_ident(ident: &tt::Ident) -> TtToken {
         SyntaxKind::from_keyword(ident.text.as_str()).unwrap_or(IDENT)
     };
 
-    TtToken { kind, is_joint_to_next: false, text: ident.text.clone(), n_tokens: 1 }
+    TtToken { kind, is_joint_to_next: false, text: ident.text.clone() }
 }
 
-fn convert_punct(p: &tt::Punct, next_tokens: &[tt::TokenTree]) -> TtToken {
-    let mut iter = next_tokens.iter();
-    iter.next();
-    let mut peek = TokenPeek::new(iter);
-
-    if let Some((kind, is_joint_to_next, text, size)) = convert_multi_char_punct(p, &mut peek) {
-        TtToken { kind, is_joint_to_next, text: text.into(), n_tokens: size }
-    } else {
-        let kind = match p.char {
-            // lexer may produce combpund tokens for these ones
-            '.' => DOT,
-            ':' => COLON,
-            '=' => EQ,
-            '!' => EXCL,
-            '-' => MINUS,
-            c => SyntaxKind::from_char(c).unwrap(),
-        };
-        let text = {
-            let mut buf = [0u8; 4];
-            let s: &str = p.char.encode_utf8(&mut buf);
-            SmolStr::new(s)
-        };
-        TtToken { kind, is_joint_to_next: p.spacing == tt::Spacing::Joint, text, n_tokens: 1 }
-    }
+fn convert_punct(p: &tt::Punct) -> TtToken {
+    let kind = match p.char {
+        // lexer may produce combpund tokens for these ones
+        '.' => DOT,
+        ':' => COLON,
+        '=' => EQ,
+        '!' => EXCL,
+        '-' => MINUS,
+        c => SyntaxKind::from_char(c).unwrap(),
+    };
```

Comment:
I wonder if we need to get rid of single-byte/multi-byte distinction in grammar.ron?

Reply:
You mean change `from_char` to support multi-bytes-able `SyntaxNode`? Or you mean remove all multi-byte `SyntaxNode`? The former is fine, but the latter will become the recent PR i did which try to handle all multibyte parser code case by case, which I don’t want to try again in recent future😂

Reply:
I mean literal merging of single and multibyte tokens in grammar.ron, and fixing up all the fallout from that :-)

It seems like, if we always use decomposed tokens, we shouldn't really care about this distinction anymore